src/onnx-mlir.cpp

```diff
@@ -58,10 +58,10 @@ int main(int argc, char *argv[]) {
 
   llvm::cl::SetVersionPrinter(getVersionPrinter);
 
-  // Parse options from argc/argv and default ONNX_MLIR_FLAG env var.
+  // Parse options from argc/argv and env var.
   llvm::cl::ParseCommandLineOptions(argc, argv,
       "ONNX-MLIR modular optimizer driver\n", nullptr,
-      OnnxMlirEnvOptionName.c_str());
+      menvVarName.c_str());
```

Comment:
I think there is a "chicken and egg" issue here. The parse command line option scan the options, but it may process the env var first (that was my understanding, but I may be wrong). So check to see the following.

```bash
export ONNX_MLIR_FLAGS="-O3"
export MY_ONNX_MLIR_FLAGS="-O0"

onnx-mlir -menvVarName= MY_ONNX_MLIR_FLAGS bibi.onnx
```

and maybe add a print of the option level just after parsing. Check that that works, namely that you get -O0.

If you get the -O3, no problem, you can always manually scan the args to search for the specific option before this call.

Reply:
If I understand it correctly, the problem you raised is that the 'menvVarName' variable will be given the default value before argv is fully processed. I have tested this and it works well. However, my guess is that there are not many args provided in the current tests. If the argv is very complicated, I wonder if the problem you mentioned will happen? I don't know. So I think manually scan the args to search for the specific option would make this safe. I will fix this then. For the custom process for the compilation, could you please create a quick PR?